src/main/java/com/coveo/spillway/Spillway.java

```diff
@@ -121,14 +121,43 @@ public boolean tryCall(T context) {
    *
    * @param context Either the name of the limit OR the object on which the propertyExtractor ({@link LimitBuilder#of(String, java.util.function.Function)})
    *                will be applied if it was specified
-   * @param cost The cost of the query
-   * @return False if one the enforced limit is exceeded, true otherwise
+   * @param cost The cost of the query, must be positive
+   * @return False if one the enforced limits is exceeded, true otherwise
    */
   public boolean tryCall(T context, int cost) {
-    return getExceededLimits(context, cost).isEmpty();
+    return getExceededLimits(context, cost, true).isEmpty();
   }
 
-  private List<LimitDefinition> getExceededLimits(T context, int cost) {
+  /**
+   * Behavess like {@link #checkLimit(Object, int)} with {@code cost} of one.
+   *
+   * @see #checkLimit(Object, int)
+   *
+   * @param context Either the name of the limit OR the object on which the propertyExtractor ({@link LimitBuilder#of(String, java.util.function.Function)})
+   *                will be applied if it was specified
+   * @return False if one the enforced limits would be exceeded, true otherwise
+   */
+  public boolean checkLimit(T context) {
+    return checkLimit(context, 1);
+  }
+
+  /**
+   * Verify if the given limit will be exceeded for the given cost
+   *
+   * @param context Either the name of the limit OR the object on which the propertyExtractor ({@link LimitBuilder#of(String, java.util.function.Function)})
+   *                will be applied if it was specified
+   * @param cost The cost of the query, must be positive
+   * @return False if one the enforced limits would be exceeded, true otherwise
+   */
+  public boolean checkLimit(T context, int cost) {
+    return getExceededLimits(context, cost, false).isEmpty();
+  }
+
+  private List<LimitDefinition> getExceededLimits(T context, int cost, boolean performAdd) {
+    if (cost < 1) {
+      throw new IllegalArgumentException("'cost' must be positive");
```

Comment:
" must be greater than 0" ? Because 0 could be considered positive, but it is an illegal value (Should be changed in the block comments also I guess)

Reply:
I guess it depends how well versed the reader is in number theory 😛